### PR TITLE
[Infra] Add Qadence2-IR, Platforms and Expressions as dependencies and examples

### PR DESCRIPTION
**Pull request by RolandMacDoland**
Remove submodules and add packages as dependencies. Add examples provided by @Doomsk.

### CODE REVIEW

File: .gitmodules
Comment (Doomsk) [inline .gitmodules:1]: if we are removing submodules, this file needs to be removed as well

**Review by Doomsk — CHANGES_REQUESTED** [review]
nothing much to do in this branch besides deciding to go with or without submodules

File: .gitmodules
Comment (RolandMacDoland) [reply .gitmodules:1]: Oh yeah that's right. Thanks.

**Review by Doomsk — APPROVED** [review]
good to me!

**Pull request merged**